mpt_trie/src/partial_trie.rs

```diff
@@ -304,6 +314,24 @@ where
 pub struct HashedPartialTrie {
     pub(crate) node: Node<HashedPartialTrie>,
     pub(crate) hash: Arc<RwLock<Option<H256>>>,
+
+    pub(crate) collapse_strategy: CollapseStrategy,
+}
+
+/// A strategy to adopt for collapsing a branch node containing only two
+/// children into an extension node after deletion of one of these two children,
+/// if the remaining child is a Hash node.
+///
+/// The default behaviour is to return an Error upon such scenario.
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
+pub enum CollapseStrategy {
+    /// Allow collapsing of single hash-node branch child into an extension
+    /// node.
+    Pass,
+    #[default]
+    /// Reject collapsing of single hash-node branch child into an extension
+    /// node.
+    Error,
 }
```

Comment:
Lovely change :) I think we could do better with the terminology though
```suggestion
/// How to handle the following subtree on deletion of the indicated node.
/// ```text
///      BranchNode
///        /    \
/// DeleteMe    OrphanedHashNode
/// ```
#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
pub enum OnDeletionOrphan {
    /// Replace `BranchNode` with an appropriate `ExtensionNode`
    CollapseToExtension,
    /// Return an error.
    #[default]
    Reject,
}
```

Reply:
Yeah I wasn't super happy with my terminology, yours is great